client/lib/store-transactions/index.js

```diff
@@ -181,27 +181,37 @@ TransactionFlow.prototype._submitWithPayment = function( payment ) {
 	};
 
 	this._pushStep( { name: SUBMITTING_WPCOM_REQUEST } );
-	wpcom.transactions( 'POST', transaction, ( error, data ) => {
-		if ( error ) {
-			return this._pushStep( {
-				name: RECEIVED_WPCOM_RESPONSE,
-				error,
-				last: true,
-			} );
-		}
+	return new Promise( resolve => {
+		wpcom.transactions( 'POST', transaction, ( error, data ) => {
+			if ( error ) {
+				this._pushStep( {
+					name: RECEIVED_WPCOM_RESPONSE,
+					error,
+					last: true,
+				} );
+				// This should probably reject but since the error is already
+				// reported just above, that might risk double-reporting or
+				// changing the step to the wrong one, so this just resolves
+				// without data instead.
+				resolve();
+				return;
+			}
 
-		if ( data.redirect_url ) {
-			return this._pushStep( {
-				name: REDIRECTING_FOR_AUTHORIZATION,
-				data: data,
+			if ( data.redirect_url ) {
+				return this._pushStep( {
```

Comment:
Shouldn't we resolve in this block as well?

Reply:
Thanks for noticing that!